pkg/etcd/etcd_test.go

```diff
@@ -0,0 +1,302 @@
+				return
+			}
+		})
+	}
+}
+
+func TestETCD_Register(t *testing.T) {
+	type args struct {
+		ctx     context.Context
+		config  *config.Control
+		handler http.Handler
+	}
+	tests := []struct {
+		name     string
+		args     args
+		setup    func(cnf *config.Control) error
+		teardown func(cnf *config.Control) error
+		wantErr  bool
+	}{
+		{
+			name: "Call Register with standard config",
+			args: args{
+				ctx:     context.TODO(),
+				config:  generateTestConfig(),
+				handler: generateTestHandler(),
+			},
+			setup: func(cnf *config.Control) error {
+				err := unit_tests.GenerateTestRuntime(cnf)
+				return err
+			},
+			teardown: func(cnf *config.Control) error {
+				unit_tests.CleanupTestDataDir(cnf)
+				return nil
+			},
+		},
+		{
+			name: "Call Register with a tombstone file created",
+			args: args{
+				ctx:     context.TODO(),
+				config:  generateTestConfig(),
+				handler: generateTestHandler(),
+			},
+			setup: func(cnf *config.Control) error {
+				if err := unit_tests.GenerateTestRuntime(cnf); err != nil {
+					return err
+				}
+				if err := os.MkdirAll(etcdDBDir(cnf), 0700); err != nil {
+					return err
+				}
+				tombstoneFile := filepath.Join(etcdDBDir(cnf), "tombstone")
+				if _, err := os.Create(tombstoneFile); err != nil {
+					return err
+				}
+				return nil
+			},
+			teardown: func(cnf *config.Control) error {
+				tombstoneFile := filepath.Join(etcdDBDir(cnf), "tombstone")
+				os.Remove(tombstoneFile)
+				unit_tests.CleanupTestDataDir(cnf)
+				return nil
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := NewETCD()
+
+			defer tt.teardown(tt.args.config)
+			if err := tt.setup(tt.args.config); err != nil {
+				t.Errorf("Setup for ETCD.Register() failed = %v", err)
+				return
+			}
+			_, err := e.Register(tt.args.ctx, tt.args.config, tt.args.handler)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ETCD.Register() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+		})
+	}
+}
+
+func TestETCD_Start(t *testing.T) {
+	type fields struct {
+		client  *etcd.Client
+		config  *config.Control
+		name    string
+		runtime *config.ControlRuntime
+		address string
+		cron    *cron.Cron
+		s3      *s3
+	}
+	type args struct {
+		ctx              context.Context
+		clientAccessInfo *clientaccess.Info
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		args     args
+		setup    func(cnf *config.Control) error
+		teardown func(cnf *config.Control) error
+		wantErr  bool
+	}{
+		{
+			name: "Start etcd without clientAccesInfo and without snapshots",
+			fields: fields{
+				config:  generateTestConfig(),
+				address: "192.168.1.123", // Local IP address
+			},
+			args: args{
+				ctx:              context.TODO(),
+				clientAccessInfo: nil,
+			},
+			setup: func(cnf *config.Control) error {
+				err := unit_tests.GenerateTestRuntime(cnf)
+				cnf.EtcdDisableSnapshots = true
```

Comment:
I think this should go on the line above and `return unit_tests.GenerateTestRuntime(cnf)` since I'm not sure you want to continue if there's an error generating the runtime.

Reply:
Fixed, point is valid